src/report-changes/core/report-changes.service.ts

```diff
@@ -88,6 +107,10 @@ export class ReportChangesService {
         tap((change: DocChangeDetails) =>
           this.checkReportConfigUpdate(change.change),
         ),
+        switchMap((change: DocChangeDetails) =>
+          // await an update of monitors for currently active reports
+          from(this.updateReportMonitors()).pipe(map(() => change)),
+        ),
```

Comment:
We could also do a deeper refactoring and remove the ChangeDetector instances completely. But the underlying bug should be fixed ASAP, so I kept that as is for now.